File: Trabajo8-Insercion/main.py
def ordenamiento_insercion(arr):
    n = len(arr)
    for i in range(1, n):
        clave = arr[i]
        j = i - 1
        
        while j >= 0 and arr[j] > clave:
            arr[j + 1] = arr[j]
            j -= 1
        arr[j + 1] = clave

# Ejemplo de uso
arreglo = [12, 11, 13, 5, 6]
ordenamiento_insercion(arreglo)

print("Arreglo ordenado:", arreglo)

File: Trabajo8-Insercion/main.js
function ordenamientoInsercion(arr) {
    const n = arr.length;
    for (let i = 1; i < n; i++) {
        let clave = arr[i];
        let j = i - 1;
        
        while (j >= 0 && arr[j] > clave) {
            arr[j + 1] = arr[j];
            j--;
        }
        arr[j + 1] = clave;
    }
}

// Ejemplo de uso
let arreglo = [12, 11, 13, 5, 6];
ordenamientoInsercion(arreglo);

console.log("Arreglo ordenado: " + arreglo.join(" "));